src/CommunityToolkit.Aspire.Hosting.EventStore/EventStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,210 @@
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Utils;
+using CommunityToolkit.Aspire.Hosting.EventStore;
+using HealthChecks.EventStore.gRPC;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding EventStore resources to the application model.
+/// </summary>
+public static class EventStoreBuilderExtensions
+{
+    private const string DataTargetFolder = "/var/lib/eventstore";
+    private const string LogTargetFolder = "/var/log/eventstore";
+
+    /// <summary>
+    /// Adds an EventStore resource to the application model. A container is used for local development.
+    /// The default image is <inheritdoc cref="EventStoreContainerImageTags.Image"/> and the tag is <inheritdoc cref="EventStoreContainerImageTags.Tag"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The port on which the EventStore endpoint will be exposed.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <example>
+    /// Add an EventStore container to the application model and reference it in a .NET project.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var eventstore = builder.AddEventStore("eventstore");
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(eventstore);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<EventStoreResource> AddEventStore(
+        this IDistributedApplicationBuilder builder, string name, int? port = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(name);
+
+        var eventStoreResource = new EventStoreResource(name);
+
+        string? connectionString = null;
+
+        builder.Eventing.Subscribe<ConnectionStringAvailableEvent>(eventStoreResource, async (@event, cancellationToken) =>
+        {
+            connectionString = await eventStoreResource.ConnectionStringExpression
+                .GetValueAsync(cancellationToken)
+                .ConfigureAwait(false)
+                ?? throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{eventStoreResource.Name}' resource but the connection string was null.");
+        });
+
+        var healthCheckKey = $"{name}_check";
+        builder.Services.AddHealthChecks()
+            .Add(new HealthCheckRegistration(
+                healthCheckKey,
+                sp => new EventStoreHealthCheck(connectionString!),
```

Comment:
I'd prefer that we pass the resource into the constructor and then resolve the connection string when the health check is executed.

If the connection string can't be resolved, the resource is unhealthy.

This way we're avoiding relying on the event and we also don't have to use the null forgiving operator.

Reply:
Actually, is that a limitation of the external health check package?

Reply:
Yes,. Unfortunately, it only takes a connection string at the moment.

Reply:
All good, I'd forgotten you were using an external library for the health check on initial review (when I wrote the first comment) and then realised at the end.